crypto/bls12381/key_bls12381.go

```diff
@@ -37,7 +39,20 @@ var _ crypto.PrivKey = &PrivKey{}
 
 type PrivKey []byte
 
-// NewPrivateKeyFromBytes build a new key from the given bytes.
+// NewPrivateKeyFromBytes generates a new random key using `secret` for the seed
+// TODO simplify once `github.com/cosmos/crypto/` has `GenPrivKeyFromSeed`
+func GenPrivKeyFromSecret(secret []byte) (PrivKey, error) {
```

Comment:
Do you mind adding a unit test for this method?

Reply:
Yes, very good point (waiting for the simplification of the function... depending on `github.com/cosmos/crypto`)

Reply:
@melekes I added 2 UTs in 2a38ef5. The second panics without the fix I did in prod code. When you are back, please check that I am _fixing_ rather than _introducing_ a bug.

Reply:
That doesn't look right. The comment explicitly says: "If msg is larger than MaxMsgLen, SHA256 sum will be signed instead of the raw bytes."

Reply:
Fixed in https://github.com/cometbft/cometbft/pull/3513/commits/affc6dfa8d105a590f9d1e7602dbe43c249a8761. See the commit message for the explanation.

Reply:
Cool, thanks for this @melekes !